### ISSUE TITLE
Errors on compile

### ISSUE THREAD

**Issue by adrian1906** [clavicule/BBTag#4]
I get the following error:

g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4 -Iinclude -I. -o tag_model.o src/core/tag_model.cpp
In file included from include/core/tag_model.h:4:0,
                 from src/core/tag_model.cpp:1:
include/core/tag_item.h:59:7: error: expected ‘;’ at end of member declaration
     ) const Q_DECL_OVERRIDE;
       ^
include/core/tag_item.h:59:13: error: ‘Q_DECL_OVERRIDE’ does not name a type
     ) const Q_DECL_OVERRIDE;
             ^
Makefile:273: recipe for target 'tag_model.o' failed
make: *** [tag_model.o] Error 1

Were you able to compile the Master branch?

I do not know enough c++ to fix the issue.

**Comment by clavicule** [comment]
Sorry for the late reply and the fact that I'm going to speculate here (I'm in a remote area with only my cell phone with me).

I believe Q_DECL_OVERRIDE was defined in Qt5 and it looks like you are building with Qt4.

I don't think I'm using any qt5 specific classes/functions, I would assume BBTag code backward compatibility with qt4 should be ok.
I think (speculating) you may just remove that macro Q_DECL_OVERRIDE from the code and it should compile and work fine.
Could you try that?

The alternative and best solution of course would be to build BBTag with Qt5.

**Comment by adrian1906** [comment]
I had installed qt5 prior to compiling however it looks like qt4 was still used when issuing the qmake command without any parameters. The error disappeared when used:
qmake -gt5 BBTag.pro

I still have an issue with gcc finding the included libraries.  For example,

the file: 
/include/ui/main_window.h
has the line:
#include<QMainWindow>

I get a compilation error saying that QMainWindow cannot be found.  Many sites say that QT += widgets in the .pro file should fix this issue.  It does not.

I also tried adding the path to the directories containing the libraries into my $PATH 
(/opt/Qt5.8.0/5.8/gcc_64/include/QtWidgets)
and
(/opt/Qt5.8.0/5.8/gcc_64/include/QtCore)

Still nothing.

My guess is that this would be a lot easier if I used qtcreator but I'm unable to load it for som reason.  I get the following errors (using Ubuntu 16.04):

libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
Unrecognized OpenGL version
Unrecognized OpenGL version

**Comment by clavicule** [comment]
That's probably the qmake from Qt4. If you type:
which qmake
It will tell you the path which probably points to the Qt4 one. You could directly use /path/to/qt5/bin/qmake

As soon as I get back to my computer I'll look up the compilation command line you have to use if you want to go that route.

In the meantime, the easiest is to launch QtCreator (normally it comes as part of the Qt distribution now, somewhere under the /path/to/qt5), load the project file and build. Normally you shouldn't hit any build issue and variables should be set properly.
Let me know how that goes.
Cheers

**Comment by clavicule** [comment]
Sorry I'm just seeing you error message about Qt creator. That's pretty weird: what command did you type to launch it?

**Comment by clavicule** [comment]
I would say, make sure the libGL.so files are under one of the path of your $LD_LIBRARY_PATH.
Could you double check that?

- Print the LD_LIBRARY_PATH
- find out where libGL.so is located (usually under /usr/lib/ somewhere 
- add the path to the variable
- launch Qt Creator
(I'll look up if there is any script to launch QtCreator that sets property variables before that)

**Comment by adrian1906** [comment]
Thanks for the info. My $LD_LIBRARY_PATH variable did not contain any values. I was able to find my libGL.so file.  I will try your suggestion and report back.

Thank you.

**Comment by clavicule** [comment]
Back to my computer: under the QtCreator directory, there should be the QtCreator binary but also a `QtCreator.sh` that will set the `LD_LIBRARY_PATH` for you, maybe that will work in your setup?
```
/path/to/qt/Tools/QtCreator/bin/qtcreator.sh
```

Here are the command lines for building without QtCreator:
```bash
cd /path/to/where/you/want/to/build/BBTag
# this could be anywhere where you have writable permissions
# it could be at the same level as BBTag, or within BBTag folder
# or anywhere else, you have the choice :)
mkdir build && cd build

# generate all the Makefile required for the compilation
# for release:
/path/to/qt/gcc_64/bin/qmake -spec linux-g++ -o Makefile /path/to/BBTag/BBTag.pro
# for debug add to command line the flags: CONFIG+=debug CONFIG+=qml_debug

# start the build
# you will see all the g++ command lines with all the flags setup for compiling and linking
make

# if everything goes well, you should see BBTag executable in the build dir
# launch it
./BBTag
```

Normally using the Makefile generated by qmake will work (qmake reads the BBTag project file - so it knows where the include and source are, which flag to set etc.).
When it comes to building a Qt App, QtCreator is just a wrapper running these commands in the background.

Hope that will work out on your end.
cheers

**Comment by adrian1906** [comment]
Thank you for making the modification. I am still unable to solve my issues with qtcreator.  I will try again tomorrow.

**Comment by clavicule** [comment]
Pretty strange. Did you get the Qt package already built or did you build from source?
By the way, did you have a chance to try the command line instructions?

**Comment by adrian1906** [comment]
I installed it by running qt-opensource-linux-x64-5.8.0.run 

Success!

I tried the command line instruction that you provided:
/opt/Qt5.8.0/5.8/gcc_64/bin/qmake -spec linux-g++ -o Makefile ../BBtag.pro CONFIG +=debug

(it looks like the qmake -qt5 command did not call this qmake directly as that one still gives errors)

Thank you for your patience in helping me and making the auto-fit change.   I am still unable to navigate past my home folder when looking for the .xml file saved in my Pictures folder. I have to enter the path explicitly.  Now that I can compile, I tried to fix it myself, but with my little knowledge of cpp, I was unable to figure it out.
